### ISSUE TITLE
Inconsistent code formatting across IDEs and code editors

### ISSUE THREAD

**Issue by nonk123** [kgsws/doom_ace#4]
Every code editor uses its own code formatting preset, but I can't figure out which one is being used here.

Proposal: use `clang-format` to force everything into the same style. The `.clang-format` file should be respected by most IDEs out there; and otherwise, it's easy enough to just run the command manually before committing.

**Comment by kgsws** [comment]
My code formatting has specific quirks which make reading code (at least for me) much faster.
I have only seen a few formatting specifications but none had quirks i use. (that's why i call it quirks)
Any ideas?

**Comment by nonk123** [comment]
Yeah, the "quirks" are exactly what I haven't been able to figure out. I've cooked up a `.clang-format` file that does similar things, but the results aren't perfect. You can check them out [here](https://github.com/nonk123/doom_ace/tree/clang-format).
